src/pynxtools/dataconverter/helpers.py

```diff
@@ -575,110 +575,52 @@ def is_value_valid_element_of_enum(value, elist) -> Tuple[bool, list]:
     return True, []
 
 
-NUMPY_FLOAT_TYPES = (np.half, np.float16, np.single, np.double, np.longdouble)
-NUMPY_INT_TYPES = (np.short, np.intc, np.int_)
-NUMPY_UINT_TYPES = (np.ushort, np.uintc, np.uint)
-# np int for np version 1.26.0
-np_int = (np.integer,)
-np_float = (np.floating,)
-# Not to be confused with `np.byte` and `np.ubyte`, these store
-# an integer of `8bit` and `unsigned 8bit` respectively.
-np_bytes = (np.bytes_,)
-np_char = (
-    np.str_,
-    np.bytes_,
-    np.chararray,
-)  # Only numpy Unicode string and Byte string
-np_bool = (np.bool_,)
-np_complex = (np.complex64, np.complex128, np.cdouble, np.csingle, np.complex_)
+nx_char = (str, np.chararray)
+nx_int = (int, np.integer)
+nx_float = (float, np.floating)
+nx_number = nx_int + nx_float
+
 NEXUS_TO_PYTHON_DATA_TYPES = {
     "ISO8601": (str,),
-    "NX_BINARY": (
-        bytes,
-        *np_bytes,
-    ),
-    "NX_BOOLEAN": (bool, *np_bool),
-    "NX_CHAR": (str, *np_char),
+    "NX_BINARY": (bytes, bytearray, np.byte, np.ubyte),
```

Comment:
I am not sure if not actually np.bytes_ is the correct one here. np.byte or np.ubyte hold single byte ints, see Rubel's comment
```suggestion
    "NX_BINARY": (bytes, bytearray, np.bytes_),
```